### PR TITLE
add kcctl conection port configurable

### PR DESCRIPTION
**Pull request by Metrora**
Signed-off-by: Liucw <liu.chuwei@99cloud.net>

### What type of PR is this?
/kind feature

### Which issue(s) this PR fixes:

Fixes https://github.com/kubeclipper-labs/kubeclipper/issues/191

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```
@x893675

### CODE REVIEW

File: cmd/kcctl/app/options/options.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ const (
 	DefaultDeployConfig       = "deploy-config.yaml"
 	DefaultConfig             = "config"
 	DefaultCaPath             = "pki"
+	DefaultSSHPort            = 22
```
Comment (x893675) [inline cmd/kcctl/app/options/options.go:68]: unnessasery

File: cmd/kcctl/app/options/options.go [hunk 2]
```diff
@@ -268,6 +269,7 @@ func NewDeployOptions() *DeployConfig {
 		IPDetect: autodetection.MethodFirst,
 		SSHConfig: &sshutils.SSH{
 			User: "root",
+			Port: DefaultSSHPort,
```
Comment (x893675) [inline cmd/kcctl/app/options/options.go:272]: 22

File: cmd/kcctl/app/options/options.go [hunk 3]
```diff
@@ -308,9 +310,15 @@ func (c *DeployConfig) MergeDeployOptions() {
 }
 
 func (c *DeployConfig) Complete() error {
+	port := 0
 	if c.Config == "" {
 		return nil
 	}
+
+	if c.SSHConfig.Port != DefaultSSHPort {
```
Comment (x893675) [inline cmd/kcctl/app/options/options.go:318]: unnessasery

File: cmd/kcctl/app/options/options.go [hunk 4]
```diff
@@ -308,9 +310,15 @@ func (c *DeployConfig) MergeDeployOptions() {
 }
 
 func (c *DeployConfig) Complete() error {
+	port := 0
```
Comment (x893675) [inline cmd/kcctl/app/options/options.go:313]: unnessasery

File: cmd/kcctl/app/options/options.go [hunk 5]
```diff
@@ -338,6 +346,12 @@ func (c *DeployConfig) Complete() error {
 		}
 	}
 
+	if port != 0 {
+		c.SSHConfig.Port = port
+	} else if c.SSHConfig.Port == 0 {
```
Comment (x893675) [inline cmd/kcctl/app/options/options.go:351]: unnessasery

**Review by lixd — APPROVED** [review]
/lgtm

**Pull request merged**